features/dd-sdk-android-trace/src/test/kotlin/com/datadog/opentracing/PendingTraceTest.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.opentracing
+
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.trace.api.sampling.PrioritySampling
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+import java.math.BigInteger
+import java.util.ConcurrentModificationException
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(value = Configurator::class)
+class PendingTraceTest {
+
+    @Test
+    fun `PendingTrace is thread safe`(forge: Forge) {
+        // Given
+        val mockTracer: DDTracer = mock()
+        whenever(mockTracer.partialFlushMinSpans) doReturn 1
+        val fakeTraceId = BigInteger.valueOf(forge.aLong())
+        val pendingTrace = PendingTrace(mockTracer, fakeTraceId)
+        val rootSpan = forge.fakeSpan(pendingTrace, mockTracer, fakeTraceId, BigInteger.ZERO, "rootSpan")
+        val countDownLatch = CountDownLatch(CONCURRENT_THREAD)
+
+        // When
+        val runnables = Array(CONCURRENT_THREAD) {
+            StressTestRunnable(pendingTrace, mockTracer, rootSpan, forge, countDownLatch)
+        }
+        runnables.forEach { Thread(it).start() }
+        countDownLatch.await(20, TimeUnit.SECONDS)
+
+        // Then
+        assertThat(countDownLatch.count).isZero()
```

Comment:
Is it necessary to assert that countdownlatch.count is zero if we're performed await? Shouldn't await block until the timeout or a count of zero?